mailserver/mailserver_test.go

```diff
@@ -162,14 +166,79 @@ func (s *MailserverSuite) TestDBKey() {
 	s.Equal(byte(i/0x1000000), k.raw[0], "big endian expected")
 }
 
+func (s *MailserverSuite) TestRequestPaginationLimit() {
+	s.setupServer(s.server)
+	defer s.server.Close()
+
+	var (
+		sentEnvelopes     []*whisper.Envelope
+		reverseSentHashes []common.Hash
+		receivedHashes    []common.Hash
+		archiveKeys       []string
+	)
+
+	now := time.Now()
+	count := uint32(10)
+
+	for i := count; i > 0; i-- {
+		sentTime := now.Add(time.Duration(-i) * time.Second)
+		env, err := generateEnvelope(sentTime)
+		s.NoError(err)
+		key, err := s.server.Archive(env)
+		s.NoError(err)
+		archiveKeys = append(archiveKeys, fmt.Sprintf("%x", key))
+		sentEnvelopes = append(sentEnvelopes, env)
+		reverseSentHashes = append([]common.Hash{env.Hash()}, reverseSentHashes...)
+	}
+
+	params := s.defaultServerParams(sentEnvelopes[0])
+	params.low = uint32(now.Add(time.Duration(-count) * time.Second).Unix())
+	params.upp = uint32(now.Unix())
+	params.limit = 6
```

Comment:
I normally avoid testing with a single value in situations like this, to avoid relying on any single value which might cause the test to pass. Maybe that's a bit paranoid and we consciously agree that we don't need to go that far, so I'd like to hear other people's opinion on that.

Reply:
@PombeirP  what would you suggest to do here?

Reply:
We could do a loop with a couple of test values (first time we pass 6, maybe the second time we pass 3). That way the numbers we use in the asserts would also not be hardcoded, but a named variable with a calculated value. However, I don't want to give you extra work if the team doesn't feel it is worth writing tests like this.

Reply:
I think it's fine for now. We can refactor it later.